supervised/automl.py

```diff
@@ -1113,3 +1233,331 @@ def from_json(self, json_data):
+    random_state : int, default=None
+        Controls the randomness of the MLjar Tuner
+
+
+    Examples
+    --------
+    Binary Classification Example:
+    >>> import pandas as pd
+    >>> from sklearn.model_selection import train_test_split
+    >>> from sklearn.metrics import roc_auc_score
+    >>> from supervised import AutoML
+    >>> df = pd.read_csv(
+    ...        "https://raw.githubusercontent.com/pplonski/datasets-for-start/master/adult/data.csv",
+    ...       skipinitialspace=True
+    ...    )
+    >>> X_train, X_test, y_train, y_test = train_test_split(
+    ... df[df.columns[:-1]], df["income"], test_size=0.25
+    ... )
+    >>> automl = AutoML()
+    >>> automl.fit(X_train, y_train)
+    >>> y_pred_prob = automl.predict_proba(X_test)
+    >>> print(f"AUROC: {roc_auc_score(y_test, y_pred_prob):.2f}%")
+
+    Multi-Class Classification Example:
+    >>> import pandas as pd
+    >>> from sklearn.datasets import load_digits
+    >>> from sklearn.metrics import accuracy_score
+    >>> from sklearn.model_selection import train_test_split
+    >>> from supervised import AutoML
+    >>> digits = load_digits()
+    >>> X_train, X_test, y_train, y_test = train_test_split(
+    ...     digits.data, digits.target, stratify=digits.target, test_size=0.25,
+    ...     random_state=123
+    ... )
+    >>> automl = AutoML(mode="Perform")
+    >>> automl.fit(X_train, y_train)
+    >>> y_pred = automl.predict(X_test)
+    >>> print(f"Accuracy: {accuracy_score(y_test, y_pred):.2f}%")
+
+    Regression Example:
+    >>> import pandas as pd
+    >>> from sklearn.datasets import load_boston
+    >>> from sklearn.model_selection import train_test_split
+    >>> from sklearn.metrics import mean_squared_error
+    >>> from supervised import AutoML
+    >>> housing = load_boston()
+    >>> X_train, X_test, y_train, y_test = train_test_split(
+    ...       pd.DataFrame(housing.data, columns=housing.feature_names),
+    ...       housing.target,
+    ...       test_size=0.25,
+    ...       random_state=123,
+    ... )
+    >>> automl = AutoML(mode="Compete")
+    >>> automl.fit(X_train, y_train)
+    >>> print("Test R^2:", automl.score(X_test, y_test))
+
+    Scikit-learn Pipeline Integration Example:
+    >>> from imblearn.over_sampling import RandomOverSampler
+    >>> from sklearn.pipeline import make_pipeline
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import train_test_split
+    >>> from supervised import AutoML
+    >>> X, y = make_classification()
+    >>> X_train, X_test, y_train, y_test = train_test_split(X,y)
+    >>> pipeline = make_pipeline(RandomOverSampler(), AutoML())
+    >>> print(pipeline.fit(X_train, y_train).score(X_test, y_test))
+    """
+
+    def __init__(
+        self,
+        mode="Explain",
+        ml_task="auto",
+        tuning_mode="Normal",
+        results_path=None,
+        total_time_limit=30 * 60,
+        model_time_limit=None,
+        algorithms="auto",
```

Comment:
I kept all `auto` parameters in `kwargs` because  I want to have a simple constructor for `AutoML`. 

The inexperienced user will find it scary to see so many parameters in the constructor.

Reply:
I get your point. But `AutoML` can also work with no params provided. So a inexperienced user can just `model = AutoML()`